sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -209,26 +216,148 @@ private HttpResponse receiveResponse(HttpRequest httpRequest, HttpURLConnection
             }
 
-            AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
-
-            try (InputStream errorStream = connection.getErrorStream();
-                 InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
-                byte[] buffer = new byte[8192];
-                int length;
-                while ((length = inputStream.read(buffer)) != -1) {
-                    outputStream.write(buffer, 0, length);
+            if (!eagerlyReadResponse) {
+                if (responseHeaders.get(HeaderName.CONTENT_TYPE).getValue().contains(TEXT_EVENT_STREAM) && listener != null) {
+                    try {
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                        processBuffer(reader);
+                    } catch (IOException e) {
+                        retryExceptionForSSE(e);
+                    }
+                }
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders);
+            } else {
+                AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
+
+                try (InputStream errorStream = connection.getErrorStream();
+                     InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
+                    byte[] buffer = new byte[8192];
+                    int length;
+                    while ((length = inputStream.read(buffer)) != -1) {
+                        outputStream.write(buffer, 0, length);
+                    }
                 }
-            }
 
-            return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
-                BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
+                    BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+            }
         } catch (IOException e) {
             throw LOGGER.logThrowableAsError(new RuntimeException(e));
+
         } finally {
             connection.disconnect();
         }
     }
 
+    /**
+     * Processes the sse buffer and dispatches the event
+     * @param reader The BufferedReader object
+     * @throws IOException If an I/O error occurs
+     */
+    private void processBuffer(BufferedReader reader) throws IOException {
+        StringBuilder sb = new StringBuilder(), data = new StringBuilder();
+        String eventStr = null;
+        long id = -1;
+        ServerSentEvent event = new ServerSentEvent();
+        int dataRead = reader.read();
+        while (dataRead != -1) {
+            sb.append((char) dataRead);
+            dataRead = reader.read();
+            int index;
+            while ((index = sb.indexOf("\n\n")) >= 0) { // blocks of data are separated by double newlines
```

Comment:
Will the server always use `\n` as a line separator? Should we check for `\r` or other newline characters?

Reply:
The standard spec [here ](https://html.spec.whatwg.org/multipage/server-sent-events.html#event-stream-interpretation )defines "\n\n"  is used to indicate the end of an event.

Reply:
I happened to see this tweet last week that might give food for thought: https://twitter.com/_clem/status/1750321943486574914